src/_pytest/unittest.py

```diff
@@ -115,6 +117,8 @@ def setup(self):
             self._request._fillfixtures()
 
     def teardown(self):
+        if self._need_tearDown:
```

Comment:
Are you sure we need this? Even if I comment these two lines, tear down still gets called for this example:

```python
from unittest import TestCase, expectedFailure

class T(TestCase):

    def tearDown(self):
        print('*********** tearDown')
        self.filename = None

    @expectedFailure
    def test_foo(self):
        self.filename = 'hello'
        assert 0

```

```
λ pytest  .tmp\foo.py -sq
*********** tearDown
x
1 xfailed in 0.05s
```

Uncommenting them actually shows tearDown being called twice:

```
 λ pytest  .tmp\foo.py -sq
*********** tearDown
x*********** tearDown

```

Reply:
Nevermind, we actually need it, but only if a test is not an expected failure